### PR TITLE
Diona Glow Bulbs

### PR DESCRIPTION
**Pull request by ParadoxSpace**
You can now weld them apart, and walk through them.

### CODE REVIEW

**Comment by ParadoxSpace** [comment]
!bugfix

**Comment by ParadoxSpace** [comment]
!review

File: code/modules/random_map/automata/diona.dm [hunk 1]
```diff
@@ -42,6 +42,16 @@
 	light_power = 3
 	light_range = 3
 	light_color = "#557733"
+	density = 0
+
+/obj/structure/diona/bulb/attackby(obj/item/W, mob/user)
+	if(W.iswelder())
+		var/obj/item/weldingtool/WT = W
+		if(WT.remove_fuel(0, user))
+			to_chat(user, "<span class='notice'>You slice through the skin of the bulb, revealing a confused nymph!</span>")
```
Comment (NonQueueingMatt) [inline code/modules/random_map/automata/diona.dm]: Not adding a do_after?

File: code/modules/random_map/automata/diona.dm [hunk 2]
```diff
@@ -42,6 +42,21 @@
...
+
+/obj/structure/diona/bulb/attackby(obj/item/W, mob/user)
+	if(W.iswelder())
+		var/obj/item/weldingtool/WT = W
+		if (!WT.welding)
+			to_chat(user, "<span class='danger'>\The [WT] must be turned on!</span>")
+		else if (WT.remove_fuel(0,user))
+			to_chat(user, "<span class='notice'>You begin slicing through the skin of \the [src].</span>")
+			if(do_after(user, 20/W.toolspeed, act_target = src))
+				if(!src || !WT.isOn()) return
```
Comment (skull132) [inline code/modules/random_map/automata/diona.dm]:
```suggestion
				if(QDELETED(src) || !WT.isOn())
					return
```

File: code/modules/random_map/automata/diona.dm [hunk 3]
```diff
@@ -42,6 +42,22 @@
 	light_power = 3
 	light_range = 3
 	light_color = "#557733"
+	density = 0
+
+/obj/structure/diona/bulb/attackby(obj/item/W, mob/user)
+	if(W.iswelder())
+		var/obj/item/weldingtool/WT = W
+		if (!WT.welding)
+			to_chat(user, "<span class='danger'>\The [WT] must be turned on!</span>")
```
Comment (skull132) [inline code/modules/random_map/automata/diona.dm]:
```suggestion
			to_chat(user, "<span class='danger'>\The [WT] must be turned on!</span>")
			return
```

File: code/modules/random_map/automata/diona.dm [hunk 4]
```diff
@@ -42,6 +42,22 @@
...
+		else if (WT.remove_fuel(0,user))
+			to_chat(user, "<span class='notice'>You begin slicing through the skin of \the [src].</span>")
+			if(do_after(user, 20/W.toolspeed, act_target = src))
+				if(QDELETED(src) || !WT.isOn())
+					return
+				playsound(src.loc, 'sound/items/Welder2.ogg', 50, 1)
+				user.visible_message("<span class='notice'>\ [user] slices through the skin of \the [src], revealing a confused diona nymph.</span>")
```
Comment (skull132) [inline code/modules/random_map/automata/diona.dm]:
```suggestion
				user.visible_message("<span class='notice'>\ [user] slices through the skin of \the [src], revealing a confused diona nymph.</span>")
			else
				return
```

**Pull request merged**